To: vim_dev@googlegroups.com
Subject: Patch 7.3.241
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.241
Problem:    Using CTRL-R CTRL-W on the command line may insert only part of
	    the word.
Solution:   Use the cursor position instead of assuming it is at the end of
	    the command. (Tyru)
Files:	    src/ex_getln.c


*** ../vim-7.3.240/src/ex_getln.c	2011-07-07 15:04:38.000000000 +0200
--- src/ex_getln.c	2011-07-07 16:38:50.000000000 +0200
***************
*** 3046,3052 ****
  	    int	    len;
  
  	    /* Locate start of last word in the cmd buffer. */
! 	    for (w = ccline.cmdbuff + ccline.cmdlen; w > ccline.cmdbuff; )
  	    {
  #ifdef FEAT_MBYTE
  		if (has_mbyte)
--- 3046,3052 ----
  	    int	    len;
  
  	    /* Locate start of last word in the cmd buffer. */
! 	    for (w = ccline.cmdbuff + ccline.cmdpos; w > ccline.cmdbuff; )
  	    {
  #ifdef FEAT_MBYTE
  		if (has_mbyte)
***************
*** 3064,3070 ****
  		    --w;
  		}
  	    }
! 	    len = (int)((ccline.cmdbuff + ccline.cmdlen) - w);
  	    if (p_ic ? STRNICMP(w, arg, len) == 0 : STRNCMP(w, arg, len) == 0)
  		p += len;
  	}
--- 3064,3070 ----
  		    --w;
  		}
  	    }
! 	    len = (int)((ccline.cmdbuff + ccline.cmdpos) - w);
  	    if (p_ic ? STRNICMP(w, arg, len) == 0 : STRNCMP(w, arg, len) == 0)
  		p += len;
  	}
*** ../vim-7.3.240/src/version.c	2011-07-07 16:20:45.000000000 +0200
--- src/version.c	2011-07-07 16:41:29.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     241,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
258. When you want to see your girlfriend, you surf to her homepage.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
